### DISCUSSION TITLE
Not accepting migrations custom path

### DISCUSSION THREAD

**Discussion by mittalyashu** [sequelize/umzug#363] [Q&A] [answered]
Even after passing a custom path (tried both ways using path module, relative path) it is settings default path as `<root>/migrations`

```js
const unzug = new Umzug({
	migrations: {
		path: "./database/migrations",
		pattern: /\.js$/
	},
	context,
	storage: new CustomStorage()
});
```

**Error**

```
TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
```

Or sometimes

```
OperationalError: ENOENT: no such file or directory, scandir './migrations'
```

**Info**
Version: v2.3.0
Sequelize: v6.3.4
Using js
Node: v12.18.3

**File structure**
- database/migrations (all db migrations files)
- database/index.js (initialise sequelize instance)
- database/umzug.js
- index.js (authenticate db)

**Comment by mmkal** [comment]
Did you try passing an absolute path to `migrations.path`? e.g. `path.join(__dirname, './migrations')`

**Comment by mittalyashu** [comment]
Yes, it said out this error

```
The "path" argument must be of type string. Received undefined
```

**Comment by mmkal** [comment]
Are you able to reproduce the error with a minimal setup? Hard to say otherwise what's wrong. Failing that, you could try adding some log statements to `node_modules/umzug/lib/index.js` in `_findMigrations()`.

**Comment by mmkal** [comment, marked as answer]
@mittalyashu it looks like you're mixing up v2 and v3 syntax. There are some breaking changes in the v3 beta. Please see the note about v2 in the readme. There are a few more changes in the pipeline, likely nothing major/breaking, but if you prefer to stick with v2 until v3 is fully stable, follow the link to the v2.x branch, or npm: https://www.npmjs.com/package/umzug/v/2.3.0
